bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/widgets/DisplayWin32Test.java

```diff
@@ -0,0 +1,55 @@
+package org.eclipse.swt.widgets;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.eclipse.swt.internal.*;
+import org.eclipse.swt.internal.win32.*;
+import org.junit.jupiter.api.*;
+
+public class DisplayWin32Test {
+
+	private Display display;
+
+	@BeforeAll
+	public static void assumeIsFittingPlatform() {
+		PlatformSpecificExecution.assumeIsFittingPlatform();
+	}
+
+	@BeforeEach
+	public void createDisplay() {
+		display = new Display();
+	}
+
+	@AfterEach
+	public void destroyDisplay() {
+		display.dispose();
+	}
+
+	@Test
+	public void setRescaleAtRuntime_activate() {
```

Comment:
NIT: I know it's not a rule but I'd rather have the tests named `test...blablabla`, it helps me to see very easily which methods are tests when using the call hierarchy.

Reply:
Can you give me a use case in which that would be helpful, i.e., why do I want to see test methods in the call hierarchy? I am always annoyed if classes have test methods starting with `test` (which is usually because these classes were originally written for JUnit 3) as it just makes the names longer and less readable without providing any benefit.

Reply:
Sure thing:

![image](https://github.com/user-attachments/assets/7e80de2f-22b0-450a-8c3e-8443419fd2a4)

In this case one can spot the last test (in green) easily and the other 3 (in red) not so easily

Reply:
I think that we have an XY problem here: I see that I can spot tests that way more easily. But I do not see _why_ I should ever want to do that. I never had a need for that.

Reply:
Not sure what _XY problem_ is but identifying the pieces of test code more easily helps me focus on the pieces of code that are using a method and the ones that are testing it. 

I personally (often) discover the meaning of methods by looking at _how_ it is used (production code) and -in an ideal world- discover what their contracts are by looking at the tests (provided that they are unit tests and that they have been created by following the TDD paradigm). 

Long story short, it helps me differentiate types of usages and prioritize where to focus my attention better.

Reply:
I am sorry, but I still don't get it. A generic "xyz helps be prioritize" will not allow me to understand use cases, in which this could be important. As mentioned: I have never felt a need for this information. In addition, I have never read about anyone propose this `test` prefix to be used a convention for whatever benefits and I am not aware of any convention for the Eclipse project but I only found it as a leftover of JUnit 3 times (and that's why there are even automated ways of getting rid of them, e.g., https://docs.openrewrite.org/recipes/java/testing/cleanup/removetestprefix).

Please provide some proper sources justifying your proposal. Otherwise, sorry to say, I will not adapt this PR to conform to a single person's personal preference.

Reply:
You never read about anyone proposing to preserve the "test" because there was no need for it: they were already there. If anything, one should read about the need to remove them (_i.e._ to change the status quo). If you ever start that discussion, please ping me ;-) 

You're reading about it now, from me.

"Leftovers" don't mean "bad", it's just legacy and legacy is not always bad. In fact, I have a use for it, even if you don't.

There are now conventions for it because, again, there was no explicit need to change/preserve the status quo.

An OpenRewrite recipe is neither a convention nor an enforcer, it's just a tool. One needs to choose when to use that tool. Conventions are specified by teams, not tools.

Again, this is not a blocker and I don't want to keep you from merging this PR. I'm just asking you to please do not get rid of something that is **useful** (to me). 

I'm not sure where do you draw the line between a "personal preference" and a "well-founded ask", but I hope you see my ask as the latter. After all, I'm not simply writing "_please don't remove them because I like them_", I'm writing "_please don't remove them because I **need** them, I find them useful_".

Reply:
I don't think this discussion leads to anything. As long as you are not willing to explain the reasons for your request, I am totally lost in finding a way to properly respond to it. Still I cannot leave some of the recent statements uncommented.

> You never read about anyone proposing to preserve the "test" because there was no need for it: they were already there. If anything, one should read about the need to remove them (_i.e._ to change the status quo). If you ever start that discussion, please ping me ;-)

The introduction of JUnit 4 made the `test` prefix obsolete, which was required in JUnit 3 times. JUnit 4 was released in 2006. Even though many projects migrated later than that, the general statement that the prefixes "were already there" is not true for almost 20 years anymore.

> You're reading about it now, from me.
> 
> "Leftovers" don't mean "bad", it's just legacy and legacy is not always bad. In fact, I have a use for it, even if you don't.
> 
> There are no conventions for it because, again, there was no explicit need to change/preserve the status quo.
> 
> An OpenRewrite recipe is neither a convention nor an enforcer, it's just a tool. One needs to choose when to use that tool. Conventions are specified by teams, not tools.

I don't see anything I wrote conflicting with this statement. Indeed, conventions are specified by people that find them useful (in a specific context). And I have already stated that I am not aware of any convention that we are following for this project. In addition, an OpenRewrite recipe is not a self purpose. Someone has written it for the sake of getting rid of test prefixes in an actual context (nor in _every_ context, of course).

> Again, this is not a blocker and I don't want to keep you from merging this PR. I'm just asking you to please do not get rid of something that is **useful** (to me).
>
> I'm not sure where do you draw the line between a "personal preference" and a "well-founded ask", but I hope you see my ask as the latter. After all, I'm not simply writing "_please don't remove them because I like them_", I'm writing "_please don't remove them because I **need** them, I find them useful_".

Maybe you are mistaken about what this PR does: I am not changing names of existing test method. I just add new test methods without that prefix, as I do not see any value in them. I never asked another PR to remove the `test` prefix in their changed/added test method. _You_ ask me to change this, without giving me the chance to understand _why_. You still did not respond to my question of explaining me (by concrete example) why your proposal is useful.

To conclude this:
- If you want to have a convention (or at least a discussion about a need for it) for this project, please start an "official" discussion in the GitHub organization, so that everyone has a chance for sharing an opinion. If others also feel the need for such a prefix, we might agree on a convention that I will gladly follow.
- If you want to convince me, explain me why your proposal makes sense (at a concrete scenario and not only because "you need it").
- Otherwise, in case there are no further objections, I propose to close this discussion and proceed with the essence of this PR and target its merge.

Reply:
👍 

Proceed.

Change the name of the class too then. It's new and JUnit doesn't need the suffix `Test` anymore either.

Reply:
> Change the name of the class too then. It's new and JUnit doesn't need the suffix `Test` anymore either.

I would really prefer to stay professional and have a serious discussion instead of making sulky statements or making fun of other people's opinions (not sure which of them was the intention here).

Reply:
> > Change the name of the class too then. It's new and JUnit doesn't need the suffix `Test` anymore either.
> 
> I would really prefer to stay professional and have a serious discussion instead of making sulky statements or making fun of other people's opinions (not sure which of them was the intention here).

I was neither trying to make fun of your opinion nor trying to be sulky.

But I see I didn't exactly hit the nail in politeness either and that my comment can be very easily misread, so please allow to me rephrase:
_it's worth considering changing the name of the class too since it falls into the same category as the name of the methods: no convention that dictates `Test` is necessary as a suffix and no need for it because of JUnit_.

Furthermore, I found out that there is one trick in Eclipse that I can use to cover my use case (distinguishing Test sources in the _hierarchy call_ view): **filters**.

![image](https://github.com/user-attachments/assets/99cb2f1c-8ea9-47ed-a109-233473bd20d3)

![image](https://github.com/user-attachments/assets/20da1bb3-c0b0-49fa-821d-9ffafd1f60b5)

With that, I can easily get rid of all calls from within _test code_, so I'm all set.

![image](https://github.com/user-attachments/assets/c1e6da8d-2342-44e6-a32b-726bf0df18bb)

On my account this PR is ready to be merged (it has been since the very beginning actually, hence the "NIT" in my 1st comment) and this discussion is starting to go sideways so there's no need to prolong it.

I thank you for your patience and I sincerely value your opinion.

Should me mark this conversation as _resolved"_? ✔️

Reply:
Yes, we can mark this as resolved. In case we need to change anything related to tests, let's do it in a follow-up to have the productive code changes in now.

Why didn't I take the last proposal serious? `org.eclipse.swt.widgets.DisplayWin32` neither sounds like a reasonable FQN for a class nor does it indicate in any way that it is a test class. It might have been okay if the class was placed in a separate package with a test indication given by some package in the FQN, but this is not the case, so it would not make any sense.